iib/workers/config.py

```diff
@@ -153,45 +159,66 @@ def validate_celery_config(conf, **kwargs):
+        'csv_annotations': {'annotations'},
+        'package_name_suffix': {'suffix'},
+        'registry_replacements': {'replacements'},
+    }
+
     for org, org_config in conf['iib_organization_customizations'].items():
         if not isinstance(org, str):
-            raise ConfigError('The keys in iib_organization_customizations must be strings')
-
-        if not isinstance(org_config, dict):
-            raise ConfigError('The values in iib_organization_customizations must be dictionaries')
-
-        invalid_keys = org_config.keys() - {
-            'csv_annotations',
-            'package_name_suffix',
-            'registry_replacements',
-        }
-        if invalid_keys:
-            raise ConfigError(
-                'The following keys set on iib_organization_customizations are '
-                f'invalid: {", ".join(sorted(invalid_keys))}'
-            )
-
-        for key in ('csv_annotations', 'registry_replacements'):
-            value = org_config.get(key)
-            if not value:
-                continue
-
-            for k, v in value.items():
-                if not isinstance(k, str):
-                    raise ConfigError(
-                        f'The keys in iib_organization_customizations.{org}.{key} must be strings'
-                    )
-
-                if not isinstance(v, str):
-                    raise ConfigError(
-                        f'The values in iib_organization_customizations.{org}.{key} must be strings'
-                    )
-
-        if not isinstance(org_config.get('package_name_suffix', ''), str):
-            raise ConfigError(
-                f'The value of iib_organization_customizations.{org}.package_name_suffix '
-                'must be a string'
+            raise ConfigError('The org keys in iib_organization_customizations must be strings')
+
+        if not isinstance(org_config, list):
+            raise ConfigError('The org values in iib_organization_customizations must be a list')
+
+        for customization in org_config:
+            if not isinstance(customization, dict):
+                raise ConfigError(
+                    'Every customization for an org in '
+                    'iib_organization_customizations must be dictionary'
+                )
+
+            customization_type = customization.get('type')
+            if customization_type not in valid_customizations.keys():
+                raise ConfigError(
+                    f'Invalid customization in iib_organization_customizations {customization}'
+                )
+
+            invalid_customization_keys = (
+                customization.keys() - valid_customizations[customization_type] - {'type'}
             )
+            if invalid_customization_keys:
+                raise ConfigError(
+                    f'The keys {invalid_customization_keys} in iib_organization_customizations'
+                    f'.{org}[{org_config.index(customization)}] are invalid.'
```

Comment:
You can just use JSON notation and use the dot. It'd be a cleaner error message. Right now you are mixing both.